types/wlr_output.c

```diff
@@ -454,6 +453,10 @@ void wlr_output_cursor_set_surface(struct wlr_output_cursor *cursor,
 		wl_signal_add(&surface->events.destroy, &cursor->surface_destroy);
```

Comment:
If I set the cursor to a different surface than the current one, do we now have listeners on both the surfaces?

Reply:
`wlr_output_cursor_reset` removes the listeners.